sound/soc/sof/ipc4-topology.h

```diff
@@ -418,6 +418,31 @@ struct sof_ipc4_src {
 	struct sof_ipc4_msg msg;
 };
 
+/**
+ * struct sof_ipc4_asrc_data - IPC data for ASRC
+ * @base_config: IPC base config data
+ * @sink_rate: Output rate for sink module
+ * @asynchronous_mode: synchronous 0, asynchronous 1. When 1 the ASRC tracks and
+ *		       compensates for drift. When 0 can test ASRC with DAI endpoint
+ *		       without timestamps capability.
+ * @operation_mode: push 0, pull 1. In push mode the  ASRC consumes a defined number
+ *		    of frames at input, with varying number of frames at output. In pull mode the
+ *		    ASRC outputs a defined number of frames while number of input frames varies
```

Comment:
You haven't replied to my question: if the number of frames can vary, then how do we define the buffer sizes in the topology. we set ibs and obs, clearly the buffer sizes or number of periods have to be set in relation with the operation mode or you could have xruns.

Reply:
Sorry, I replied as this https://github.com/thesofproject/linux/pull/4557#discussion_r1325290511.
"at this time, there is 2 internal circular buffers of ASRC itself to store the input/output samples. and the buffer size is 10 frames bigger than a period samples."
 And I haven't see any reply, so I thought you accepted it.

Reply:
Humm, this is very odd to me.

Why is 10 a good value? This looks completely arbitrary and a poor choice IMHO. Why use e,g, 58 samples per period instead of a more computer-friendly value of 64....

This should be determined by 
- the amount of variability expected in the rate
- the amount the tracking algorithm can deal with
-  circular buffer management
 
and this should be a factor of the period size, not a fixed increment, If our Chrome friends use a 10ms period, 10 samples doesn't look so useful...

And this buffering should absolutely be documented in code comments. Hiding such fundamental design decisions is not good at all. You are introducing a whole new class of algorithms, and this should be done by clearly explaining all assumptions.